catho/quantum
# feat(QTM-713): Input component migrated to CSS modules

## Description
https://jirasoftware.catho.com.br/browse/QTM-713

## Review guide
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] TypeScript updated

### How to test build

- In quantum, install the dependencies and build, running `yarn && yarn build`
- Clone the repository https://github.com/catho/ssr-quantum-simulator
- In ssr-quantum-simulator, add the local quantum build using `yarn add [local-path]/quantum/dist`
- Change the GlobalStyle imports to Typography in the _app and quantum.js files.
- In the _app file, import the quantum css using `import '@catho/quantum/index.css'`
- Still in ssr-quantum-simulator, run `yarn dev` and access the page http://localhost:3000/quantum
- Compare components with https://catho.github.io/quantum/?path=/docs/introduction-gettingstarted--docs

### How to test visual changes
- In quantum, install the dependencies and build, running `yarn && yarn storybook`
- Access https://catho.github.io/quantum/?path=/docs/introduction-gettingstarted--docs and http://localhost:9007/, in parallel, and make visual comparisons of the components affected by the changes. 

Note: 
- It is important to check states (hover, focus, disabled, etc.), as they are not valid for regressive tests (yarn test:regression)
- Changes that are not related to the objective of the task are related to the merge made from the master branch. The intention here is to keep all features from the master branch in the beta branch
